src/canvas.py

```diff
@@ -215,10 +215,10 @@ def set_legend(self):
                     new_lines, labels,
                     loc=self.application.plot_settings.legend_position,
                     frameon=True, reverse=True)
-            else:
-                legend = self.top_right_axis.get_legend()
-                if legend is not None:
-                    legend.remove()
+        else:
```

Comment:
This change reintroduces the bug, where deleting items won't make it disappear.
```suggestion
                return
        legend = self.top_right_axis.get_legend()
        if legend is not None:
            legend.remove()
```